unison-cli/src/Unison/Codebase/Transcript/Runner.hs

```diff
@@ -204,12 +205,23 @@ run isTest verbosity dir codebase runtime sbRuntime nRuntime ucmVersion baseURL
           -- We shorten the terminal width, because "Transcript" manages a 2-space indent for output lines.
           Pretty.toPlain (terminalWidth - 2) line
 
-      maybeDieWithMsg :: String -> IO ()
+      maybeDieWithMsg :: Pretty.Pretty Pretty.ColorText -> IO ()
       maybeDieWithMsg msg = do
-        errOk <- readIORef allowErrors
-        if errOk
-          then writeIORef hasErrors True
-          else dieWithMsg msg
+        liftIO $ writeIORef hasErrors True
+        liftIO (liftA2 (,) (readIORef allowErrors) (readIORef expectFailure)) >>= \case
+          (False, False) -> liftIO . dieWithMsg $ Pretty.toPlain terminalWidth msg
+          (True, True) ->
+            transcriptFailure
+              out
+              ( Text.pack . Pretty.toPlain terminalWidth $
+                  "The stanza above previously had an incorrect successful result, but now fails with"
```

Comment:
my brain melts with "incorrect successful result"
```suggestion
                  "The stanza above previously had a successful result due to a known bug, but the behavior has changed and it now fails with"
```

Reply:
I think we should come up with some consistent terms, e.g.:
- “succeeded” – had a non-error result, regardless of whether that was intended
- “errored” – had an error result, regardless of whether that was intended
- “passed” – had the intended result, regardless of whether it was success or error
- “failed” – had an unintended result, regardless of whether it was success or error (or just a different error than expected)

Doesn’t have to be those terms in particular, but I think the succeeded/errored pair should be complementary terms, and the passed/failed pair should also be complementary.

“Succeeded” is the only term there that I’m not particularly happy with.

In this case, the stanza already has `:error :bug` on it, and we want to suggest the removal of `:bug`. My original version doesn’t satisfy my terminology above because it says “fail” when it should say “error”. Maybe something like, “The stanza above is now erroring with […] as expected. Please remove `:bug`.”?